### PR TITLE
[record-minmax] Save minmax to circle

### PR DESCRIPTION
**Pull request by jinevening**
This derives minmax values and saves them to cicle model

ONE-DCO-1.0-Signed-off-by: Hyukjin Jeong <hj1.jeong@samsung.com>

---
Related to #1605

### CODE REVIEW

File: compiler/record-minmax/src/RecordMinMax.cpp [hunk 1]
```diff
@@ -32,6 +36,38 @@ using DataType = luci_interpreter::DataType;
...
+/**
+ * @brief  clipAndAverage does the following things
+ *         (1) Discard the top/bottom clip_ratio (%) values from vector
+ *             Ex: top_clip_ratio = 5, bottom_clip_ratio = 3
+ *                 discard top 5 % values and bottom 3 % values from vector
+ *         (2) Calculate the average of remaining values
+ */
+float clipAndAverage(std::vector<float> &vector, float top_clip_ratio, float bottom_clip_ratio)
+{
+  if (bottom_clip_ratio < 0 || bottom_clip_ratio >= 100)
```
Comment (seanshpark) [inline compiler/record-minmax/src/RecordMinMax.cpp:48]: `>= 100` will not allow 100. typo? or intended?

File: compiler/record-minmax/src/RecordMinMax.cpp [hunk 2]
```diff
@@ -131,15 +167,32 @@ void RecordMinMax::profileData(const std::string &input_data_path)
...
-  (void *)minmax_map;
+  for (auto iter = minmax_map->begin(); iter != minmax_map->end(); ++iter)
+  {
+    auto node = iter->first;
+    auto minmax = iter->second;
+
+    // Default: Values are averaged without clipping (clip ratio = 0 %).
+    // TODO: Allow users to adjust clip ratios.
+    float min =
+        clipAndAverage(minmax.min_vector, /* top_clip_ratio */ 0.0, /* bottom_clip_ratio */ 0.0);
```
Comment (seanshpark) [inline compiler/record-minmax/src/RecordMinMax.cpp:178]: this will get 0.0 ~ 0.0 ?

File: compiler/record-minmax/src/RecordMinMax.cpp (hunk 1 above)
Comment (jinevening) [reply compiler/record-minmax/src/RecordMinMax.cpp:48]: It was intended. If clip_ratio is 100 %, all data will be clipped, so we cannot record min/max values.

File: compiler/record-minmax/src/RecordMinMax.cpp (hunk 2 above)
Comment (jinevening) [reply compiler/record-minmax/src/RecordMinMax.cpp:178]: Clip ratio is the ratio of clipped values. So, 0.0 means that we use all data. If top_clip_ratio is 1.0, top 1 % values will be discarded. If bottom_clip_ratio is 1.0, bottom 1% values will be discarded.

File: compiler/record-minmax/src/RecordMinMax.cpp [hunk 3]
```diff
@@ -32,6 +36,38 @@ using DataType = luci_interpreter::DataType;
...
+ * @brief  clipAndAverage does the following things
+ *         (1) Discard the top/bottom clip_ratio (%) values from vector
+ *             Ex: top_clip_ratio = 5, bottom_clip_ratio = 3
+ *                 discard top 5 % values and bottom 3 % values from vector
+ *         (2) Calculate the average of remaining values
+ */
+float clipAndAverage(std::vector<float> &vector, float top_clip_ratio, float bottom_clip_ratio)
+{
+  if (bottom_clip_ratio < 0 || bottom_clip_ratio >= 100)
+    throw std::runtime_error("Clip ratio (bottom) must be between 0 and 100");
```
Comment (seanshpark) [inline compiler/record-minmax/src/RecordMinMax.cpp:49]: > It was intended. 

Message `0 and 100` does not correspond with your intention. I think it sould be like `between 0 and 99`

Comment (jinevening) [reply compiler/record-minmax/src/RecordMinMax.cpp:49]: Clip ratio is float, so I simply used 0 and 100 (99.1 is allowed).

Accurate expression would be "Clip ratio must be greater than or equal to 0 and less than 100.". I will change the message. Thanks.

File: compiler/record-minmax/src/RecordMinMax.cpp [hunk 4]
```diff
@@ -131,15 +169,32 @@ void RecordMinMax::profileData(const std::string &input_data_path)
   }
 
   auto minmax_map = _observer->minMaxData()->getMap();
-  (void *)minmax_map;
+  for (auto iter = minmax_map->begin(); iter != minmax_map->end(); ++iter)
```
Comment (mhs4670go) [inline compiler/record-minmax/src/RecordMinMax.cpp:172]: (trivial)
```suggestion
  for (auto &it : minmax_map)
```

Comment (seanshpark) [reply compiler/record-minmax/src/RecordMinMax.cpp:172]: @jinevening , do you want to apply this proposal or want to merge as-is?

Comment (jinevening) [reply compiler/record-minmax/src/RecordMinMax.cpp:172]: @seanshpark Can you merge this as-is? I will make the change later.

**Pull request merged**